### PR TITLE
Prepare 2021 July Release [HLS 1.3.0]

### PR DESCRIPTION
**Pull request by Ailrun**
CC: @jneira, @berberman, @pepeiborra, @isovector

Resolves partially: #1990.

### CODE REVIEW

**Review by jneira — APPROVED** [review]
It is nice to see all merged prs at once
Many thanks for take care of release, including version bumps! ❤️

**Comment by jneira** [comment]
> Resolves partially: #1990.

why partially?

**Comment by Ailrun** [comment]
@jneira Because this PR itself does not finish the release process, as there are more things like making sure binary builds well, GHCUp coordination, Hackage release, ...

**Comment by jneira** [comment]
Yeah, the pr is the starting point. Your plan is make the hackage release close to the github one?

**Comment by Ailrun** [comment]
Yes, I think managing single-release-related things in a single issue is better for modularity

File: ChangeLog.md [hunk 1]
```diff
@@ -1,5 +1,153 @@
 # Changelog for haskell-language-server
 
+## 1.3.0
+
+2021 July release of HLS arrives! This release includes binaries for GHC 9.0.1
+and some new interesting features. Here is the brief summary of changes:
+
+- Binaries for GHC 9.0.1 are added by @anka-213.
+- Call hierarchy plugin is added, contributed by @July541.
+  ![hierarchy](https://user-images.githubusercontent.com/12473268/127550041-094151a6-be7b-484a-bb82-c61f326ca503.gif)
```
Comment (Ailrun) [inline ChangeLog.md:10]: Now, I notice that call hierarchy does not work when execute it on a type signature. Is this intended, @July541?

**Comment by jneira** [comment]
I started to work in a gh job to automate the hackage release, i hope i will able to finish it some day 😅 

https://github.com/jneira/haskell-language-server/blob/hackage-release/.github/workflows/hackage.yml

File: ChangeLog.md [hunk 2]
```diff
@@ -1,5 +1,153 @@
...
+- Now completions work with definitions from non-imported modules, thanks to @pepeiborra.
+  ![completion](https://user-images.githubusercontent.com/12473268/127543694-718ae043-38f2-4fb0-be71-317f5f93b443.gif)
+- Eval plugin
+  - The plugin supports GHC 9.0.1, thanks to @berberman.
+  - `:info` command is added by @akrmn.
+  - The plugin uses the same default language as GHCi with @fmehta's patch.
+- Wingman, where most changes owing to @isovector
+  - An option for disabling proof state styling is added.
+  - Tactic supports deep recursion.
+  - Hole fit suggestions are now disabled with Wingman for a performance reason.
```
Comment (isovector) [inline ChangeLog.md:20]:
```suggestion
  - Hole fit suggestions are now disabled for performance reasons when using Wingman. 
```

File: ChangeLog.md [hunk 3]
```diff
@@ -1,5 +1,153 @@
...
+  ![hierarchy](https://user-images.githubusercontent.com/12473268/127550041-094151a6-be7b-484a-bb82-c61f326ca503.gif)
+- Now completions work with definitions from non-imported modules, thanks to @pepeiborra.
+  ![completion](https://user-images.githubusercontent.com/12473268/127543694-718ae043-38f2-4fb0-be71-317f5f93b443.gif)
+- Eval plugin
+  - The plugin supports GHC 9.0.1, thanks to @berberman.
+  - `:info` command is added by @akrmn.
+  - The plugin uses the same default language as GHCi with @fmehta's patch.
+- Wingman, where most changes owing to @isovector
+  - An option for disabling proof state styling is added.
+  - Tactic supports deep recursion.
```
Comment (isovector) [inline ChangeLog.md:19]:
```suggestion
```

File: ChangeLog.md [hunk 4]
```diff
@@ -1,5 +1,153 @@
...
+- Call hierarchy plugin is added, contributed by @July541.
+  ![hierarchy](https://user-images.githubusercontent.com/12473268/127550041-094151a6-be7b-484a-bb82-c61f326ca503.gif)
+- Now completions work with definitions from non-imported modules, thanks to @pepeiborra.
+  ![completion](https://user-images.githubusercontent.com/12473268/127543694-718ae043-38f2-4fb0-be71-317f5f93b443.gif)
+- Eval plugin
+  - The plugin supports GHC 9.0.1, thanks to @berberman.
+  - `:info` command is added by @akrmn.
+  - The plugin uses the same default language as GHCi with @fmehta's patch.
+- Wingman, where most changes owing to @isovector
+  - An option for disabling proof state styling is added.
```
Comment (isovector) [inline ChangeLog.md:18]:
```suggestion
  - Wingman no longer changes the fixity of function definitions.
  - Wingman now gives unique names to the holes it generates.
  - Significant improvements in Wingman's ability to reason about polymorphic and GADT types.
  - Wingman no longer suggests homomorphic destructs when the codomain is larger than the domain.
  - "Complete case constructors" support for empty lambda cases
  - Wingman will now give a warning if it ran out of gas during "attempt to fill hole"
  - Metaprogramming improvements, including symbolic-name support, and the `pointwise` combinator.
  - Fixed a bug when attempting to introduce too many variables in an `intros` metaprogram.
```

**Review by isovector — APPROVED** [review]
lgtm! thanks for tackling this, @Ailrun !

File: ChangeLog.md (hunk 4 above)
Comment (Ailrun) [reply ChangeLog.md]: May I ask why you want to remove the line for proof styling option?

Besides that,

> - Fixed a bug when attempting to introduce too many variables in an `intros` metaprogram.

should these be highlighted?

File: ChangeLog.md [hunk 5]
```diff
@@ -1,5 +1,159 @@
...
+  - The plugin uses the same default language as GHCi with @fmehta's patch.
+- Wingman, where most changes owing to @isovector
+  - Wingman no longer changes the fixity of function definitions.
+  - Wingman now gives unique names to the holes it generates.
+  - Significant improvements in Wingman's ability to reason about polymorphic and GADT types.
+  - Wingman no longer suggests homomorphic destructs when the codomain is larger than the domain.
+  - "Complete case constructors" support for empty lambda cases
+  - Wingman will now give a warning if it ran out of gas during "attempt to fill hole"
+  - Metaprogramming improvements, including symbolic-name support, and the `pointwise` combinator.
+  - Fixed a bug when attempting to introduce too many variables in an `intros` metaprogram.
```
Comment (Ailrun) [inline ChangeLog.md:25]:
```suggestion
  - Wingman's ability to reason about polymorphic and GADT types is significantly improved.
  - Wingman no longer suggests homomorphic destructs when the codomain is larger than the domain.
  - "Complete case constructors" action supports empty lambda cases.
  - Wingman now gives a warning if it ran out of gas during "attempt to fill hole"
  - Metaprogramming for Wingman has been improved with symbolic-name support and the `pointwise` combinator.
```

How do you think, @isovector?

**Review by pepeiborra — APPROVED** [review]

**Review by berberman — APPROVED** [review]

File: ChangeLog.md (hunk 1 above)
Comment (July541) [reply ChangeLog.md:10]: Yes, reference other languages' implementations, it doesn't belong to call hierarchy I think.

**Comment by July541** [comment]
I'm confused why not release LSP first? LSP has update about ghc 9.0 and call hierarchy, in case we can remove at least the depencency in cabal.project.

**Comment by jneira** [comment]
we can make the release in GitHub and upload binaries cause the build uses the cabal.project
But thanks to note it, cause we can't do the *hackage* release without having all dependencies in hackage as well

File: ChangeLog.md (hunk 4 above)
Comment (isovector) [reply ChangeLog.md]: > May I ask why you want to remove the line for proof state styling option?

it's the least interesting thing in the change, imo. does anyone even know what a proof state is?

File: ChangeLog.md (hunk 5 above)
Comment (isovector) [reply ChangeLog.md:25]: :+1:

**Pull request merged**